Doc/library/devmode.rst

```diff
@@ -0,0 +1,184 @@
+.. _devmode:
+
+Python Development Mode
+=======================
+
+.. versionadded:: 3.7
+
+The Python Development Mode introduces additional runtime checks which are too
+expensive to be enabled by default. It should not be more verbose than the
+default if the code is correct: new warnings are only emitted when an issue is
+detected.
+
+It can be enabled using :option:`-X dev <-X>` command line option or by setting
+:envvar:`PYTHONDEVMODE` environment variable to ``1``.
+
+Effects of the developer mode
+=============================
+
+Enabling the development mode has multiple effects:
+
+* Add ``default`` :ref:`warning filter <describing-warning-filters>`, as
+  :option:`-W default <-W>` command line option. The following warnings are
+  shown, whereas they are ignored by the default warning filters:
+
+  * :exc:`DeprecationWarning`;
+  * :exc:`ImportWarning`;
+  * :exc:`PendingDeprecationWarning`;
+  * :exc:`ResourceWarning`.
+
+  Use :option:`-W error <-W>` command line option or set
+  :envvar:`PYTHONWARNINGS` environment variable to ``error`` to treat warnings
+  as errors.
+
+* Install debug hooks on memory allocators to check for:
+
+  * buffer underflow;
+  * buffer overflow
+  * memory allocator API violation;
+  * usage of the GIL.
```

Comment:
I'm not sure any of these should end in a semi-colon, but if you want to keep that style, you're missing one.  :-)

Reply:
I've seen this style used before in a few different areas throughout the documentation, so I think it's okay. But, it should definitely be consistent within the same listing of bullet points. (: